src/runtime/extensiontype.cs

```diff
@@ -50,8 +50,9 @@ public ExtensionType()
         public static void FinalizeObject(ManagedType self)
         {
             Runtime.PyObject_GC_Del(self.pyHandle);
-            Runtime.XDecref(self.tpHandle);
-            self.gcHandle.Free();
+            // Not necessary decref for `tpHandle`.
+            // Runtime.XDecref(self.tpHandle);
+            self.FreeGCHandle();
```

Comment:
`PyObject_GC_Del` is used for deallocate the unmanaged memory, GCHandle is used for holding the managed object to make it alive when no clr reference, since you're referring the `ManagedType self`, the self won't be a garbage.